### ISSUE TITLE
Petition to add sub-termini/multi-termini

### ISSUE THREAD

**Issue by Vectorial1024** [klyte45/TransportLinesManager#59]
Right now TLM provides a convenient way to identify and relocate transport line termini/starting stop, and this feature is heavily utilized by Express Bus Services (EBS) (https://github.com/Vectorial1024/ExpressBusServices) to determine termini bus stops; it makes sense to allow unbunching at termini to reasonably simulate real life bus operations.

Right now, some EBS users have petitioned to EBS about allowing them to set multiple termini along the route for better simulation (as documented here https://github.com/Vectorial1024/ExpressBusServices/issues/7). I very much understand and agree with their sentiments, but find the addition of multi-termini selection hard to do, due to me not having the expertise of UI modding, and the need to cater for at least 2 versions of the GUI: vanilla GUI and TLM GUI. (IPT2 is not included here because EBS already have integrated with IPT2 to read multi-termini configurations.)

----

Later, as I discover this GitHub repo, I noticed a discussion very similar to what EBS is trying to do (https://github.com/klyte45/TransportLinesManager/issues/37), and discussion began to shift to the addition of sub-termini config for TLM so that TLM and Write The Signs can benefit from the added info to be displayed on various props.

If sub-termini selection were to be added to TLM, EBS would benefit as well for being able to read into TLM config and decide what stop to unbunch etc. At the same time, with termini selection available, TLM would be able to directly compete with IPT2 on public transportation: IPT2 allows unbunching control, but lacks fine control on fare and frequency, and TLM + EBS would provide both unbunching control, fine control on fare and frequency, and (+ Write The Signs) deeper immersion.

Would be nice to see this implemented.

Declaration of interests: I am the author of Express Bus Services.

**Comment by klyte45** [comment]
Well, I'm thinking to make it different: make bus buildings with stops to be recognized as terminus and make them to unbunch in there also. In TLM 14, the unbunching process will be done only at first stop, which is selected in the world info panel of the line.

Anyway, I'll think better which philosophy will fit better to set the terminus. Tram and trolleybus also have express mode, but no stops buildings like the buses, so maybe a stop selection would be more consistent thought all the transport types.

**Comment by GameBurrow** [comment]
> Well, I'm thinking to make it different: make bus buildings with stops to be recognized as terminus 

People don't always use bus buildings for start and end, it's getting quite common to build "custom" terminuses using ingame roads. Also in my game I have multiple lines that pass for example the international bus terminal. but that isn't the terminus for that line, so I would prefer to have full control myself on what stops are the terminuses.

**Comment by klyte45** [comment]
Yep, this is another point. The building oriented solution is getting weaker, almost discarded. Let we see if I can add this before the TLM 14 deadline.

**Comment by Vectorial1024** [comment]
> Well, I'm thinking to make it different: make bus buildings with stops to be recognized as terminus and make them to unbunch in there also.

Let Design = buses always unbunch at bus termini buildings.

----

As a left-hand drive (LHD) CSL player, I strongly oppose the Design because of the nightmare blockage that will be unleashed. This is because many many bus termini buildings (even in vanilla game) are designed to be RHD friendly, which is equivalent to them being LHD unfriendly. The Design will cause nightmarish blockage that turns LHD players off.

I demonstrate the point using a simplified vanilla Bus Terminus (After Dark).

In RHD, bus ops at the terminal will look like this:

<img width="734" alt="RHD" src="https://user-images.githubusercontent.com/17726797/115348498-4446b780-a1e5-11eb-98e5-5b0ee7b643e2.png">

Everything works, everyone is happy, no bus line color needed to show the point.

Now, what it means to be LHD in City Skylines is simply that all roads are reversed. This already resulted in a nightmarish bus ops in the same terminal in LHD:

<img width="734" alt="LHD" src="https://user-images.githubusercontent.com/17726797/115349078-fbdbc980-a1e5-11eb-93d4-3361363c936f.png">

Nothing works, no one is happy. Whenever some bus lines unbunch, there must exist some other bus line that is affected by blocking the entrance.

> it's getting quite common to build "custom" terminuses using ingame roads.

As a LHD player this is almost my only accepted solution to bus termini due to the above LHD blockage nightmare. Only for the looks would I _begin_ to consider placing some termini assets.

-----

And, as a mod player myself, I still mildly oppose the Design.

There is this mod Advanced Stop Selection (https://steamcommunity.com/sharedfiles/filedetails/?id=1394468624), that allows players to place stops in multiplatform buildings that are otherwise impossible. It's originally designed for multiplatform metro stations and train stations, but the working principle makes the mod also affect bus termini buildings, so I could theoretically set up some unusual termini config: (again, same simplified Bus Termini)

<img width="734" alt="Mod" src="https://user-images.githubusercontent.com/17726797/115350132-4c9ff200-a1e7-11eb-9ce8-7c7e4164e8f7.png">

And then it will cause blockage. This blockage will be fully player-made, but still, unbunching like this will create problems even for a RHD terminus.

--- 

So really, will be much better to have separate controls instead, not from design philosophy, but for practical reasons.

**Comment by klyte45** [comment]
Well, it will be implemented as part of express buses because it will be need in some buildings, exactly because what you explained. In some models the path to the stop is so short that the bus skips the stop in anyway. To fix this, will be mandatory the stop to be marked as it must stop there, exactly what the feature discussed here will bring to the simulation in the express buses. 

Adding to v14 milestone.

**Comment by klyte45** [comment]
Details about the implementation:

1. Only road types lines can have stops marked as terminal stop.
2. The first stop will always be a terminal, like it was when the express lines were implemented.
3. All terminals will be marked with a star in the line representations:
![image](https://user-images.githubusercontent.com/11671356/115955237-c0821900-a4cb-11eb-8bcc-1d2e67d8e69e.png)
4. Use right click to set or unset a stop as terminal (except first stop)
5. Auto name will take all terminals with a heavier value. But if they are so close each other, it try to use a stop in the other peer of line.
6. Auto name will still showing up to 2 destinations, but all them will be used in the Write The Signs, and will change the destination variable value when it reach a terminal stop. 
So when I did some changes to the example line, note the line name changed, but the Airport destination still showing up in the buses after reaching the _Cap-de-Ile_ terminal.
While stop names don't use districts as name, the target destinations do use if selected as allowed to use in auto name. This explains why the address stop became _Cap-de-Ile_ in the destination (the district of the terminal stop). A custom name directly set in the stop will be used, if exists.

This bus have just left the New Pixel train station stop. Was targeting the Airport, now goes to Cap-de-Ile:
![image](https://user-images.githubusercontent.com/11671356/115955505-3b97ff00-a4cd-11eb-8282-b630212f9188.png)

Right after reaching that station, it start targeting the Airport:
![image](https://user-images.githubusercontent.com/11671356/115955549-831e8b00-a4cd-11eb-9e91-4513bf6cf586.png)

And in the airport, it became targeting the New Pixel train station:
![image](https://user-images.githubusercontent.com/11671356/115955566-9cbfd280-a4cd-11eb-9cc4-b8dce7053252.png)

It's already in alpha branch. Use the WTS 0.3 r3 to get the new multiple destination feature.
